alerts/cluster-monitoring-operator/ClusterMonitoringOperatorDeprecatedConfig.md

```diff
@@ -0,0 +1,33 @@
+# ClusterMonitoringOperatorDeprecatedConfig
+
+## Meaning
+
+This alert fires when a deprecated config is used in the config map `openshift-monitoring/cluster-monitoring-config`.
+
+## Impact
+
+Using a deprecated config is not recommended because the config has no effect and
+it may cause `cluster-monitoring-operator` go in degraded state when the config
+is removed in a future release.
+
+## Diagnosis
+
+- Check the __Description__ and __Summary__ sections of the alert to identify
+the
+  deprecated config as in the following example:
+
+  __Description__
+
+  `The configuration field k8sPrometheusAdapter in
+  openshift-monitoring/cluster-monitoring-config was deprecated in 4.16 and has
+  no effect, you can remove it.`
+
+  __Summary__
+
+  `Cluster Monitoring Operator is being used with deprecated configuration.`
+
+## Mitigation
+
```

Comment:
I think it's better if we handle each situation individually:
- For the `k8sPrometheusAdapter.dedicatedServiceMonitors` field, you can remove the block, see the release notes (entry) at XXX (for clusters <4.16)
- For the `k8sPrometheusAdapter` field, check the release notes entry at YYY, some of the subfields may need to be migrated under `metricsServer` (or sth like this)